### PR TITLE
chore: fix doc gen dependency graph

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing build commands and task configurations across various JSON files, particularly for the `vercel.json` and `package.json` files. It introduces new build scripts, modifies existing ones, and updates task dependencies.

### Detailed summary
- Added `buildCommand` in `examples/ui-demo/vercel.json`.
- Updated `installCommand` and `buildCommand` in `site/vercel.json`.
- Introduced new `build` task in `examples/ui-demo/turbo.json`.
- Renamed build step in `.github/workflows/on-pull-request.yml` to `Build Libraries`.
- Updated `run` command to `yarn build:libs` in the same workflow.
- Modified `docs:gen` command to use `node` instead of `npx` in multiple `package.json` files.
- Changed `build` command in the main `package.json` to `yarn build:libs`.
- Introduced `build:demo`, `build:site`, and `build:examples` scripts in the main `package.json`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
* **#1146** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @moldy530 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: site/vercel.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
+  "installCommand": "cd .. && yarn install --ignore-engines",
+  "buildCommand": "cd .. && yarn build:base",
```
Comment (moldy530) [inline site/vercel.json:3]: just wanna make sure all of this runs from the root. I will look into updating the settings to always operate from root and correctly deploy the files from `site/dist` later

File: package.json [hunk 2]
```diff
@@ -26,10 +26,11 @@
     "postgenerate": "yarn lint:write",
     "preinstall": "yarn config set ignore-engines true",
     "postinstall": "git submodule update --init --recursive && patch-package",
-    "build": "is-ci && yarn build:ci || yarn build:dev",
-    "build:base": "turbo run build --filter=!embedded-accounts-quickstart --filter=!ui-demo",
-    "build:dev": "yarn build:base --filter=!docs",
-    "build:ci": "yarn build:base && yarn lint:write",
+    "build": "yarn build:libs",
```
Comment (moldy530) [inline package.json:29]: this is for backwards compat, I guess I could just merge this and `build:libs`, but I wanted the separation just to be explicit in the CI script

**Comment by moldy530** [comment]
hmmmm the test failure here is passing locally -.-

**Review by dphilipson — DISMISSED** [review]

**Pull request merged**